riak_test/src/rtcs.erl

```diff
@@ -885,55 +935,25 @@ error_child_element_verifier(Code, Message, Resource) ->
             true
     end.
 
-%% @doc TODO: FIXME: XXX: once we move to 2.0.x series, we stop using
-%% this function and go back to rt_cs_dev:upgrade/2. This is just
-%% copy&paste and modify.
-%% This also resets config and moves to default CS riak config.
-upgrade_20(Node, NewVersion) ->
-    N = rt_cs_dev:node_id(Node),
-    Version = rt_cs_dev:node_version(N),
-    lager:info("Upgrading ~p : ~p -> ~p", [Node, Version, NewVersion]),
-    catch rt_cs_dev:stop(Node),
-    rt:wait_until_unpingable(Node),
-    OldPath = rt_cs_dev:relpath(Version),
-    NewPath = rt_cs_dev:relpath(NewVersion),
-
-    Commands = [
-                io_lib:format("cp -v -p -P -R \"~s/dev/dev~b/data\" \"~s/dev/dev~b\"",
-                              [OldPath, N, NewPath, N]),
-                io_lib:format("rm -rf ~s/dev/dev~b/data/*",
-                              [OldPath, N]),
-                io_lib:format("cp \"~s/dev/dev~b/etc/app.config\" \"~s/dev/dev~b/etc/advanced.config\"",
-                              [OldPath, N, NewPath, N])
-               ],
-    [ begin
-        lager:info("Running: ~s", [Cmd]),
-        os:cmd(Cmd)
-    end || Cmd <- Commands],
-    VersionMap = orddict:store(N, NewVersion, rt_config:get(rt_versions)),
-    rt_config:set(rt_versions, VersionMap),
-    rt_cs_dev:start(Node),
-    rt:wait_until_pingable(Node),
-    ok.
-
 %% @doc update current app.config, assuming CS is already stopped
 upgrade_cs(N, AdminCreds) ->
     migrate_cs(previous, current, N, AdminCreds).
 
-%% @doc copy and update config file from `From' to `To' version.
+%% @doc update config file from `From' to `To' version.
 migrate_cs(From, To, N, AdminCreds) ->
     migrate(From, To, N, AdminCreds, cs).
 
 migrate(From, To, N, AdminCreds, Who) when
       (From =:= current andalso To =:= previous)
       orelse ( From =:= previous andalso To =:= current) ->
-    Config = read_config(From, N, Who),
     Prefix = get_rt_config(Who, To),
+    Config = get_config(Who, To),
```

Comment:
This is for multibag flavor. mb test dopes mb configuration.. so `read_config` salvages those configurations. There might be other special configuration change involved. This `read_config` is for them.